### PR TITLE
TELCODOCS#1876: Moved troubleshooting assembly modules to the main LVMS assembly

### PR DESCRIPTION
**Pull request by sr1kar99**
Version(s):
4.13+

Issue:
[TELCODOCS-1876](https://issues.redhat.com/browse/TELCODOCS-1876)

Link to docs preview:
[Troubleshooting persistent storage](https://75839--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.html#lvms-troubleshooting-persistent-storage_logical-volume-manager-storage)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by sr1kar99** [comment]
@jakobmoellerdev 
Could you please review this PR?
Thanks!

File: modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc [hunk 1]
```diff
@@ -4,17 +4,20 @@
 
 :_mod-docs-content-type: PROCEDURE
 [id="investigating-a-pvc-stuck-in-the-pending-state_{context}"]
-= Investigating a PVC stuck in the Pending state
+= Investigating a PVC stuck in the pending state
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc:7]:
```suggestion
= Investigating a PVC stuck in the `Pending` state
```

Pending is a specific state that is observable, which is why its upper cased

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 2]
```diff
@@ -37,23 +41,23 @@ $ oc get logicalvolume
...
 +
 [source,terminal]
 ----
 $ oc patch logicalvolume <name> -p '{"metadata":{"finalizers":[]}}' --type=merge <1>
 ----
-<1> Replace `<name>` with the name of the CR.
+<1> Replace `<name>` with the name of the `LogicalVolume` CR.
 
-.. After removing their finalizers, delete the CRs by running the following command:
+.. After removing the finalizers, delete the `LogicalVolume` CRs by running the following command:
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:52]: This is incorrect I believe, one first needs to trigger a delete of the LogicalVolumes, and then remove finalizers as necessary

@suleymanakbas91 please correct me if wrong here

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 3]
```diff
@@ -66,30 +70,30 @@ $ oc get lvmvolumegroup
...
 ----
-<1> Replace `<name>` with the name of the CR.
+<1> Replace `<name>` with the name of the `LVMVolumeGroup` CR.
 
-. Remove any `LVMVolumeGroupNodeStatus` CRs by running the following command:
+. Delete any `LVMVolumeGroupNodeStatus` CRs by running the following command:
 +
 [source,terminal]
 ----
 $ oc delete lvmvolumegroupnodestatus --all
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:93]: I think all objects (except LogicalVolume here) are namespaced, meaning it should be run with a project configured or with a --namespace

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc [hunk 4]
```diff
@@ -4,13 +4,18 @@
 
 :_mod-docs-content-type: PROCEDURE
 [id="recovering-from-missing-lvms-or-operator-components_{context}"]
-= Recovering from missing LVMS or Operator components
+= Recovering from missing storage class
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:7]:
```suggestion
= Recovering from missing StorageClass
```

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc [hunk 5]
```diff
@@ -67,9 +48,14 @@ topolvm-node-dr26h                    4/4     Running   0             66m
 vg-manager-r6zdv                      1/1     Running   0             66m
 ----
 +
-The expected output is one running instance of `lvms-operator` and `vg-manager`. One instance of `topolvm-controller` and `topolvm-node` is expected for each node.
+The output of this command must contain a running instance of the following pods:
+
+* `lvms-operator`
+* `vg-manager`
+* `topolvm-controller`
+* `topolvm-node`
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:56]:
```suggestion
```
This is no longer the case for 4.16 and beyond, but still necessary for lower versions

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc [hunk 6]
```diff
@@ -67,9 +48,14 @@ topolvm-node-dr26h                    4/4     Running   0             66m
...
 +
-If `topolvm-node` is stuck in the `Init` state, there is a failure to locate an available disk for LVMS to use. To retrieve the information necessary to troubleshoot, review the logs of the `vg-manager` pod by running the following command:
+If the `topolvm-node` pod is stuck in the `Init` state, it is due to a failure to locate an available disk for {lvms} to use. To retrieve the necessary information to troubleshoot this issue, review the logs of the `vg-manager` pod by running the following command:
```
Comment (jakobmoellerdev) [inline modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:58]:
```suggestion
If the `vg-manager` pod is stuck trying to load a configuration file, it is due to a failure to locate an available disk for {lvms} to use. To retrieve the necessary information to troubleshoot this issue, review the logs of the `vg-manager` pod by running the following command:
```

This needs to change to check for vg-manager being stuck for 4.16+

File: modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc (hunk 1 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc:7]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc (hunk 4 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:7]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc (hunk 6 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:58]: As there are two updates specific to 4.16+, I will make note of these and create another PR to make these updates only in 4.16.

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc (hunk 5 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc:56]: As there are two updates specific to 4.16+, I will make note of these and create another PR to make these updates only in 4.16.

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc (hunk 3 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:93]: The first step in this proc is:

Switch to the `openshift-storage` namespace by running the following command:
`$ oc project openshift-storage`

Considering this step, will the other commands without "-n <namespace" work fine?

**Comment by sr1kar99** [comment]
@radeore 
Can you please review this PR?
Thanks!

**Comment by radeore** [comment]
@sr1kar99 Other than existing suggested changes, LGTM!

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc (hunk 2 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc]: Updated the procedure as follows:

1. Check if the `LogicalVolume` custom resources (CRs) are present by running the following command:
+
[source,terminal]
----
$ oc get logicalvolume
----
+
.Example output
[source,terminal]
----
No resources found
----

a. If the `LogicalVolume` CRs are present, delete them by running the following command:
+
[source,terminal]
----
$ oc delete logicalvolume <name> <1>
----
<1> Replace `<name>` with the name of the `LogicalVolume` CR.

b. After deleting the `LogicalVolume` CRs, remove their finalizers by running the following command:
+
[source,terminal]
----
$ oc patch logicalvolume <name> -p '{"metadata":{"finalizers":[]}}' --type=merge <1>
----
<1> Replace `<name>` with the name of the `LogicalVolume` CR.

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc (hunk 3 above)
Comment (jakobmoellerdev) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:93]: Ah my bad, yes with that in mind, we are good here

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 7]
```diff
@@ -37,23 +41,23 @@ $ oc get logicalvolume
...
 [source,terminal]
 ----
-$ oc patch logicalvolume <name> -p '{"metadata":{"finalizers":[]}}' --type=merge <1>
+$ oc delete logicalvolume <name> <1>
 ----
-<1> Replace `<name>` with the name of the CR.
+<1> Replace `<name>` with the name of the `LogicalVolume` CR.
 
-.. After removing their finalizers, delete the CRs by running the following command:
+.. After deleting the `LogicalVolume` CRs, remove their finalizers by running the following command:
```
Comment (radeore) [inline modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:52]:
```suggestion
.. Before deleting the `LogicalVolume` CRs, remove their finalizers by running the following command:
```

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 8]
```diff
@@ -66,30 +70,30 @@ $ oc get lvmvolumegroup
...
 [source,terminal]
 ----
-$ oc patch lvmvolumegroup <name> -p '{"metadata":{"finalizers":[]}}' --type=merge <1>
+$ oc delete lvmvolumegroup <name> <1>
 ----
-<1> Replace `<name>` with the name of the CR.
+<1> Replace `<name>` with the name of the `LVMVolumeGroup` CR.
 
-.. After removing their finalizers, delete the CRs by running the following command:
+.. After deleting the `LVMVolumeGroup` CRs, remove their finalizers by running the following command:
```
Comment (radeore) [inline modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:81]:
```suggestion
.. Before deleting the `LVMVolumeGroup` CRs, remove their finalizers by running the following command:
```

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc (hunk 2 above)
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc]: @radeore - But I believe @jakobmoellerdev  had a different perspective on this, if I am not wrong.

Comment (radeore) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc]: @sr1kar99 Oh right, my bad.  Removing finalizers step should be performed after deleting the resources.

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 9]
```diff
@@ -66,30 +70,30 @@ $ oc get lvmvolumegroup
...
-. Remove any `LVMVolumeGroupNodeStatus` CRs by running the following command:
+. Delete any `LVMVolumeGroupNodeStatus` CRs by running the following command:
 +
 [source,terminal]
 ----
 $ oc delete lvmvolumegroupnodestatus --all
 ----
 
-. Remove the `LVMCluster` CR by running the following command:
+. Delete the `LVMCluster` CR by running the following command:
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:96]: Updated as per the suggestion.

Should we also add a step to remove `LVMVolumeGroupNodeStatus` CR finalizers?
@radeore

[the remark this answers is not in the document]
Comment (radeore) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:84]: Not required for LVMVolumeGroupNodeStatus resources.

[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:84]: @jakobmoellerdev @radeore 
Thanks for the clarification. Could you please add the lgtm label if the updates look good :)

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 10]
```diff
@@ -4,15 +4,19 @@
...
 
-. All of the persistent volume claims (PVCs) created using the logical volume manager storage (LVMS) driver have been removed.
+* You have installed the {oc-first}.
 
-. The pods using those PVCs have been stopped.
+* You have logged in to the {oc-first} as a user with `cluster-admin` permissions.
+
+* You have deleted all the persistent volume claims (PVCs) created using {lvms}.
+
+* You have stopped the pods that are using the PVCs created using {lvms}.
```
Comment (mburke5678) [inline modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc:19]: Same issue here with {lvms}

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc [hunk 11]
```diff
@@ -222,3 +222,36 @@ include::modules/lvms-download-log-files-and-diagnostics.adoc[leveloffset=+1]
...
+
+* xref:../../../storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc#about-creating-lvmcluster-cr_logical-volume-manager-storage[Ways to create an LVMCluster custom resource]
+
+include::modules/lvms-troubleshooting-recovering-from-node-failure.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+[id="additional-resources-forced-cleanup-1"]
+.Additional resources
+
+* xref:../../../troubleshooting-local-persistent-storage-using-lvms.adoc#performing-a-forced-cleanup_logical-volume-manager-storage[Performing a forced clean-up]
```
Comment (mburke5678) [inline storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc:247]: This link is not working for me in the Preview.

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc [hunk 12]
```diff
@@ -222,3 +222,36 @@ include::modules/lvms-download-log-files-and-diagnostics.adoc[leveloffset=+1]
...
+
+include::modules/lvms-troubleshooting-recovering-from-disk-failure.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+[id="additional-resources-forced-cleanup-2"]
+.Additional resources
+
+* xref:../../../troubleshooting-local-persistent-storage-using-lvms.adoc#performing-a-forced-cleanup_logical-volume-manager-storage[Performing a forced clean-up]
```
Comment (mburke5678) [inline storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc:255]: This link is not working for me in the Preview.

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc (hunk 11 above)
Comment (mburke5678) [reply storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc:247]: Is this it?
```suggestion
* xref:../../..//storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.html#performing-a-forced-cleanup_logical-volume-manager-storage[Performing a forced clean-up]
```

**Comment by mburke5678** [comment]
@sr1kar99 A few comments. Otherwise LGTM

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc (hunk 11 above)
Comment (sr1kar99) [reply storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc]: Fixed this link:

It should have been: 
"* xref:../../../storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc#performing-a-forced-cleanup_logical-volume-manager-storage[Performing a forced clean-up]"

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc (hunk 12 above)
Comment (sr1kar99) [reply storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc]: Fixed this link:

It should have been: 
"* xref:../../../storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc#performing-a-forced-cleanup_logical-volume-manager-storage[Performing a forced clean-up]"

File: modules/lvms-troubleshooting-recovering-from-node-failure.adoc [hunk 13]
```diff
@@ -6,7 +6,14 @@
 [id="recovering-from-node-failure_{context}"]
 = Recovering from node failure
 
-Sometimes a persistent volume claim (PVC) is stuck in a `Pending` state because a particular node in the cluster has failed. To identify the failed node, you can examine the restart count of the `topolvm-node` pod. An increased restart count indicates potential problems with the underlying node, which may require further investigation and troubleshooting.
+A persistent volume claim (PVC) can be stuck in the `Pending` state due to a node failure in the cluster. 
+
+To identify the failed node, you can examine the restart count of the `topolvm-node` pod. An increased restart count indicates potential problems with the underlying node, which may require further investigation and troubleshooting.
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-node-failure.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc [hunk 14]
```diff
@@ -4,15 +4,19 @@
...
 
 .Prerequisites
 
-. All of the persistent volume claims (PVCs) created using the logical volume manager storage (LVMS) driver have been removed.
+* You have installed the {oc-first}.
 
-. The pods using those PVCs have been stopped.
+* You have logged in to the {oc-first} as a user with `cluster-admin` permissions.
+
+* You have deleted all the persistent volume claims (PVCs) created using {lvms}.
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-performing-a-forced-cleanup.adoc]: Updated as per the suggestion.

**Review by jakobmoellerdev — APPROVED** [review]
/lgtm

File: modules/lvms-troubleshooting-recovering-from-disk-failure.adoc [hunk 15]
```diff
@@ -6,7 +6,27 @@
...
+The following table describes the volume failure error messages:
+
+.Volume failure error messages
+[%autowidth, options="header"]
+|===
+
+|Error message |Description
+|`Failed to check volume existence` |Indicates a problem in verifying whether the volume already exists. Volume verification failure can be caused by network connectivity problems or other failures.
+|`Failed to bind volume` |Failure to bind a volume can happen if the persistent volume (PV) that is available does not match the requirements of the PVC.
+|`FailedMount or FailedUnMount` |This error indicates problems when trying to mount the volume to a node or unmount a volume from a node. If the disk has failed, this error can appear when a pod tries to use the PVC.
```
Comment (jeana-redhat) [inline modules/lvms-troubleshooting-recovering-from-disk-failure.adoc:22]: Is this one error message with an "or" in it, or is it two possible messages (like so:)?
```suggestion
|`FailedMount` or `FailedUnMount` |This error indicates problems when trying to mount the volume to a node or unmount a volume from a node. If the disk has failed, this error can appear when a pod tries to use the PVC.
```

**Review by jeana-redhat — COMMENTED** [review]
You made some really nice enhancements here while moving the content! I've left a few comments about mostly meta things to take a look at. This is really nice though :)

/remove-label merge-review-in-progress
/remove-label merge-review-needed

File: modules/lvms-troubleshooting-recovering-from-disk-failure.adoc [hunk 16]
```diff
@@ -16,18 +36,10 @@ If you see a failure message while inspecting the events associated with the per
...
-- *Volume is already exclusively attached to one node and cannot be attached to another:* This error can appear with storage solutions that do not support `ReadWriteMany` access modes.
+<1> Replace `<pvc_name>` with the name of the PVC.
 
 . Establish a direct connection to the host where the problem is occurring.
 
 . Resolve the disk issue.
 
-After you have resolved the issue with the disk, you might need to perform the forced cleanup procedure if failure messages persist or reoccur.
\ No newline at end of file
+If the volume failure messages persist or recur even after you have resolved the issue with the disk, you must perform a forced clean-up. For more information, see "Performing a forced clean-up".
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-disk-failure.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-disk-failure.adoc [hunk 17]
```diff
@@ -6,7 +6,27 @@
...
+.Volume failure error messages
+[%autowidth, options="header"]
+|===
+
+|Error message |Description
+|`Failed to check volume existence` |Indicates a problem in verifying whether the volume already exists. Volume verification failure can be caused by network connectivity problems or other failures.
+|`Failed to bind volume` |Failure to bind a volume can happen if the persistent volume (PV) that is available does not match the requirements of the PVC.
+|`FailedMount or FailedUnMount` |This error indicates problems when trying to mount the volume to a node or unmount a volume from a node. If the disk has failed, this error can appear when a pod tries to use the PVC.
+|`Volume is already exclusively attached to one node and cannot be attached to another` |This error can appear with storage solutions that do not support `ReadWriteMany` access modes.
+|===
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-disk-failure.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-node-failure.adoc [hunk 18]
```diff
@@ -30,5 +37,5 @@ vg-manager-r6zdv                      1/1     Running   0             66m
 vg-manager-990ut                      1/1     Running   0             66m
 vg-manager-an118                      1/1     Running   0             66m
 ----
-+
-After you resolve any issues with the node, you might need to perform the forced cleanup procedure if the PVC is still stuck in a `Pending` state.
\ No newline at end of file
+
+If the PVC is stuck in the `Pending` state even after you have resolved any issues with the node, you must perform a forced clean-up. For more information, see "Performing a forced clean-up".
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-node-failure.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc [hunk 19]
```diff
@@ -4,13 +4,18 @@
 
 :_mod-docs-content-type: PROCEDURE
 [id="recovering-from-missing-lvms-or-operator-components_{context}"]
-= Recovering from missing LVMS or Operator components
+= Recovering from missing StorageClass
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-missing-lvms-or-operator-components.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-investigating-a-pvc-stuck-in-the-pending-state.adoc:1]: Oh yes! I missed this. 
Updated as per the suggestion.

File: modules/lvms-troubleshooting-persistent-storage.adoc [hunk 20]
```diff
@@ -0,0 +1,9 @@
+// Module included in the following assemblies:
+//
+// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc
+
+:_mod-docs-content-type: PROCEDURE
```
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-persistent-storage.adoc]: Updated as per the suggestion.

File: modules/lvms-troubleshooting-recovering-from-disk-failure.adoc (hunk 16 above)
[the remark this answers is not in the document]
Comment (sr1kar99) [reply modules/lvms-troubleshooting-recovering-from-disk-failure.adoc]: Updated as per the suggestion.

**Comment by sr1kar99** [comment]
> You made some really nice enhancements here while moving the content! I've left a few comments about mostly meta things to take a look at. This is really nice though :)
> 
> /remove-label merge-review-in-progress /remove-label merge-review-needed



> You made some really nice enhancements here while moving the content! I've left a few comments about mostly meta things to take a look at. This is really nice though :)
> 
> /remove-label merge-review-in-progress /remove-label merge-review-needed

Hi @jeana-redhat 
Thank you for the review! :) 
Updated the content as per the suggestions. Could you PTAL?
Thanks! :)

**Comment by jeana-redhat** [comment]
> > You made some really nice enhancements here while moving the content! I've left a few comments about mostly meta things to take a look at. This is really nice though :)
> > /remove-label merge-review-in-progress /remove-label merge-review-needed
> 
> > You made some really nice enhancements here while moving the content! I've left a few comments about mostly meta things to take a look at. This is really nice though :)
> > /remove-label merge-review-in-progress /remove-label merge-review-needed
> 
> Hi @jeana-redhat Thank you for the review! :) Updated the content as per the suggestions. Could you PTAL? Thanks! :)

Yep! on it now

**Review by jeana-redhat — APPROVED** [review]
Updates LGTM, thanks and great work!

**Comment by jeana-redhat** [comment]
@sr1kar99 I have merged 4.16, unfortunately looks like the rest need manual intervention

**Comment by sr1kar99** [comment]
> @sr1kar99 I have merged 4.16, unfortunately looks like the rest need manual intervention

Hi @jeana-redhat 
I have cherry-picked to the other branches. Could you PTAL and merge those PRs?
Thanks! :)

**Pull request merged**